### PR TITLE
[Environments] Multi account bug & smaller fixes

### LINKED ISSUE #3100: When logged in both my MSA and Azure AD account, I could not see any Dev Box from my Azure AD account any more
### Dev Home version

_No response_

### Windows build number

_No response_

### Other software

_No response_

### Steps to reproduce the bug

1. Login both Azure AD and MSA Account on Dev Home
2. Go to environment tab and hit refresh
3. No Dev Box ever shows up any more

### Expected result

_No response_

### Actual result

image

### Included System Information

_No response_

### Included Extensions Information

_No response_

### LINKED ISSUE #3390: Environments page sync button gets a little cut off when resizing
### Dev Home version

_No response_

### Windows build number

_No response_

### Other software

_No response_

### Steps to reproduce the bug

Open the Environments page and resize horizontally.

### Expected result

The button shouldn't cut off

### Actual result

image

### Included System Information

_No response_

### Included Extensions Information

_No response_

### LINKED ISSUE #3490: Environment provider headers still show up in the UI on environments page even if there are no environments for that provider
### Dev Home version

_No response_

### Windows build number

_No response_

### Other software

_No response_

### Steps to reproduce the bug

remove all hyper-v VMs or WSL distributions. Then navigate to the environments page. You'll see this:
image

### Expected result

There should be no headers shown for providers who don't have any environments available.

### Actual result

Headers are shown even for providers without environments to show

### Included System Information

_No response_

### Included Extensions Information

_No response_

### PR DESCRIPTION
**Pull request by huzaifa-d**
## Summary of the pull request
This PR addresses:
1. Fixes the bug not showing environments/VMs when multiple users are signed in and one of the accounts has errors.
2. Removes legacy debug only code
3. Resizes elements so that they aren't cutoff while resizing the app
4. Reverts the grouping on page, as this is no longer required

## Validation steps performed
Checked each individual change manually

## PR checklist
- [ ] Closes #3100 
- [ ] Closes #3390 
- [ ] Closes #3490

### CODE REVIEW

File: tools/Environments/DevHome.Environments/Extensions/ServiceExtensions.cs [hunk 1]
```diff
@@ -17,9 +17,6 @@ public static IServiceCollection AddEnvironments(this IServiceCollection service
         // View-models
         services.AddSingleton<LandingPageViewModel>();
 
-        // Services
```
Comment (huzaifa-d) [inline tools/Environments/DevHome.Environments/Extensions/ServiceExtensions.cs:20]: This was only used for a Debug only testing scenario. Removing as it is now out of date.

File: tools/Environments/DevHome.Environments/Selectors/CardItemTemplateSelector.cs [hunk 2]
```diff
@@ -11,7 +11,7 @@ public class CardItemTemplateSelector : DataTemplateSelector
 {
     public DataTemplate? CreateComputeSystemOperationTemplate { get; set; }
 
-    public DataTemplate? PerProviderComputeSystemTemplate { get; set; }
+    public DataTemplate? ComputeSystemTemplate { get; set; }
```
Comment (huzaifa-d) [inline tools/Environments/DevHome.Environments/Selectors/CardItemTemplateSelector.cs:15]: PerProviderComputeSystemTemplate ->  ComputeSystemTemplate
These changes are all part of the "Grouping" revert : Environments page recently got a grouping of VMs added to the UI, but this was later disapproved by the designer. These changes are from the revert commit.

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs [hunk 3]
```diff
@@ -18,6 +18,7 @@
 using DevHome.Common.Services;
```
Comment (huzaifa-d) [inline tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:18]: All the changes on this page are part of the "Grouping" revert, except for small changes to the SearchHandler and adding enum to SortHandler.

File: tools/Environments/DevHome.Environments/Views/LandingPage.xaml [hunk 4]
```diff
@@ -173,42 +173,6 @@
                 </Grid>
             </DataTemplate>
 
-            <!-- Per Provider Card Template -->
-            <DataTemplate x:Key="PerProviderComputeSystemTemplate" x:DataType="vm:PerProviderViewModel">
```
Comment (huzaifa-d) [inline tools/Environments/DevHome.Environments/Views/LandingPage.xaml:177]: PerProviderComputeSystemTemplate -> ComputeSystemTemplate
These changes are all part of the "Grouping" revert.

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs [hunk 5]
```diff
@@ -240,30 +243,25 @@ public async Task LoadModelAsync(bool useDebugValues = false)
...
 
-        foreach (var providerViewModel in PerProviderViewModels)
+        lock (ComputeSystemCards)
         {
-            var computeSystemCards = providerViewModel.ComputeSystems;
-            lock (computeSystemCards)
+            for (var i = ComputeSystemCards.Count - 1; i >= 0; i--)
             {
-                for (var i = computeSystemCards.Count - 1; i >= 0; i--)
+                if (ComputeSystemCards[i] is ComputeSystemViewModel computeSystemViewModel)
```
Comment (dhoehna) [inline tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:250]: NIT: ComputeSystemCards.Where(x => x in ComputerSystemViewModel).Foreach(x => x.RemoveStateChangedHandler(); x.ComputeSystemErrorReceived -= OnComputeSystemOperationError;).Remove();

Maybe.  I don't know if .Remove can be appended to this list.  I don't have the code up to test.

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs [hunk 6]
```diff
@@ -329,32 +319,31 @@ private async Task AddAllComputeSystemsFromAProvider(ComputeSystemsLoadedData da
...
-            _log.Error($"No Compute systems found for provider: {provider.Id}");
-        }
 
-        // Initialize the cards for the compute systems in parallel before adding them to the view model on UI thread
-        var packageFullName = data.ProviderDetails.ExtensionWrapper.PackageFullName;
+        // List of ComputeSystemViewModels to be added to the view model
+        // that didn't have any errors during initialization
         var computeSystemViewModels = new List<ComputeSystemViewModel>();
-        foreach (var computeSystem in computeSystemList)
+        foreach (var mapping in data.DevIdToComputeSystemMap.Where(map =>
```
Comment (dhoehna) [inline tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:326]: Yay LINQ!

**Review by dhoehna — APPROVED** [review]

File: tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs [hunk 7]
```diff
@@ -391,32 +380,34 @@ private async Task AddAllComputeSystemsFromAProvider(ComputeSystemsLoadedData da
...
-
-        var dontShowIndices = new List<int>();
-        for (var i = 0; i < currentProviders.Count; i++)
+        ComputeSystemCardsView.Filter = system =>
         {
-            var providerViewModel = currentProviders[i];
-            providerViewModel.SearchHandler(query);
-
-            if (!providerViewModel.IsVisible)
+            if (system is CreateComputeSystemOperationViewModel createComputeSystemOperationViewModel)
```
Comment (sshilov7) [inline tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs:385]: <Not for this PR as it's a revert of a previous change>
I think we abuse easiness that C# gives us to determine variable type at runtime (reflection) and use in places where we could do a better design instead. In this case we could have an interface that "system" would be required to implement in order to be used for search queries. It's "system's" job to know how to match that query. Then the code here would just be "return system.Match(_selectedProvider, query);"

**Review by sshilov7 — APPROVED** [review]

**Pull request merged**